src/pybkgmodel/model.py

```diff
@@ -9,79 +9,178 @@
 
 from pybkgmodel.camera import RectangularCameraImage
 
-
-def runwise_wobble_map(target_run, runs, xedges, yedges, energy_edges, cuts='None', time_delta=0.2*u.hr, pointing_delta=2*u.deg):
-    neighbours = find_run_neighbours(target_run, runs, time_delta, pointing_delta)
+class BaseMap:
     
-    if MagicEventFile.is_compatible(target_run.file_name):
-            evtfiles = [
-                MagicEventFile(run.file_name, cuts=cuts)
-                for run in (target_run,) + neighbours
-            ]
-    elif LstEventFile.is_compatible(target_run.file_name):
-            evtfiles = [
-                LstEventFile(run.file_name, cuts=cuts)
-                for run in (target_run,) + neighbours
-            ]
-    else:
-        raise RuntimeError(f"Unsupported file format for '{target_run.file_name}'.")
+    def __init__(self):
+        pass
+
+    def get_runwise_bkg(self):
+        pass
+
+    def check_data_format(self, target_run, neighbours):
```

Comment:
Name is confusing: this method creates the list of event files instead of checking the data format (though such check is performed internally). Suggest renaming to `read_runs()` or something similar.

Reply:
Name changed following your suggestion.